### DISCUSSION TITLE
Exclude many directories?

### DISCUSSION THREAD

**Discussion by aronchick** [BurntSushi/ripgrep#2790] [Q&A] [answered]
I've been trying just about everything i can find, but when I execute:
```
rg vim ~/.* # Looking for any dot files that include vim
```

with the following section in my .ripgreprc
```
...
--glob=!**/.git/
--glob=!**/.rustup/
--glob=!**/.vscode/
--glob=!**/.pyenv/
--glob=!**/.virtualenvs/
--glob=!**/.npm/
--glob=!**/.rbenv/
--glob=!**/.cursor/
--glob=!**/.cache/
--glob=!**/.bun/
```

I still get searches in .virtualenvs (among other places). What am I missing? Is there a cleaner way to exclude lots of directories?

**Comment by BurntSushi** [comment]
I think that you probably want `!**/.virtualenvs/**`. Directory exclusion probably doesn't work how you think because ripgrep still needs to descend into directories like `.virtualenvs` in case a subsequent rule whitelists something inside of `.virtualenvs`.

**Reply by aronchick** [reply]
thanks! is there a better way to have a list for exclusion? having tons of flags seems like i'm holding it wrong

**Reply by BurntSushi** [reply, marked as answer]
Yes. Use `.rgignore` or `.ignore`.

**Reply by aronchick** [reply]
Thanks!
